packages/browser/lib/utils/datatypes.ts

```diff
@@ -15,28 +17,21 @@ export const isNumber = (data: unknown): data is number => typeof data === "numb
 export const isBoolean = (data: unknown): data is boolean => typeof data === "boolean";
 export const isFile = (data: unknown): data is (File | Blob) => data instanceof File || data instanceof Blob;
 
-export const isEncryptable = (data: unknown): boolean =>
+export const isEncryptableAsString = (data: unknown): data is EncryptableAsString =>
   isDefined(data) && (isString(data) || isNumber(data) || isBoolean(data));
 
-export const getHeaderType = (data: unknown): string | undefined => {
-  if (!isDefined(data)) return;
+export const getHeaderType = (data: NonNullable<unknown>): string => {
   if (isArray(data)) return data.constructor.name;
   else {
     return typeof data;
   }
 };
 
-export const ensureString = (data: unknown): string | undefined => {
-  // TODO: the ensureString function... is not garanteed to return a string 
-  if (isUndefined(data)) return;
-
-  if (!isDefined(data)) return JSON.stringify(data);
-  if (isString(data)) return data.trim();
-  if (["bigint", "function"].includes(typeof data)) {
-    return data.toString();
-  }
-
-  return JSON.stringify(data);
+export const ensureString = (data: EncryptableAsString): string => {
+  if(isString(data)) return data;
+  else if(isNumber(data)) return data.toString();
+  else if(isBoolean(data)) return data.toString();
+  else throw new Error(`Cannot ensure string for ${data}`);
```

Comment:
I'm not conviced on why we are handling bigint or function cocersion here. However, since I'm not yet giving rigorus type defenitiions for the encrypt function yet, I'll put it back for now.